core/common/src/IModelError.ts

```diff
@@ -8,16 +8,24 @@
 
 import {
   BentleyError, BentleyStatus, BriefcaseStatus, ChangeSetStatus, DbResult, GetMetaDataFunction, IModelStatus, RepositoryStatus,
-  RpcInterfaceStatus,
 } from "@itwin/core-bentley";
 
-export { BentleyStatus, BentleyError, IModelStatus, BriefcaseStatus, GetMetaDataFunction, LogFunction, DbResult, RepositoryStatus, ChangeSetStatus, RpcInterfaceStatus } from "@itwin/core-bentley";
+export {
+  BentleyStatus, BentleyError, IModelStatus, BriefcaseStatus, GetMetaDataFunction, LogFunction, DbResult, ChangeSetStatus,
+} from "@itwin/core-bentley";
+
+/** Numeric values for common errors produced by iTwin.js APIs, typically provided by [[IModelError]].
+ * The values within each of these `enum`s are guaranteed not to conflict with one another.
+ * @public
+ */
+export type IModelErrorNumber = IModelStatus | DbResult | BentleyStatus | BriefcaseStatus | ChangeSetStatus;
```

Comment:
why not union number too instead of doing it later?

Reply:
Because then the guarantee of "no collisions" goes away. `IModelError` is-a `BentleyError` for which `errorNumber` can be any numeric value; these `enum`s are specific kinds of numbers.